### PR TITLE
React 18. Mark deprecated API

### PR DESCRIPTION
**Pull request by turansky**
[Details](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#deprecations)

cc @sgrishchenko @aerialist7

### CODE REVIEW

File: kotlin-react-dom/src/main/kotlin/react/dom/hydrate.kt [hunk 1]
```diff
@@ -6,8 +6,10 @@ package react.dom
 import org.w3c.dom.Element
 import react.ReactElement
 
-// See https://reactjs.org/docs/react-dom.html
-
+/**
+ * [Deprecation details](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#deprecations)
+ */
+@Deprecated("Will be removed soon!")
```
Comment (aerialist7) [inline kotlin-react-dom/src/main/kotlin/react/dom/hydrate.kt:12]: But this is not a truth. React team will not remove it soon.

**Review by aerialist7 — CHANGES_REQUESTED** [review]

File: kotlin-react-dom/src/main/kotlin/react/dom/hydrate.kt (hunk 1 above)
Comment (Leonya) [reply kotlin-react-dom/src/main/kotlin/react/dom/hydrate.kt:12]: I agree, let's mark as deprecated, but not as planned for removal.

Comment (turansky) [reply kotlin-react-dom/src/main/kotlin/react/dom/hydrate.kt]: Done

**Comment by Leonya** [comment]
Pushed my version here: https://github.com/JetBrains/kotlin-wrappers/commit/adadade46e4c55d9098fed4f32f1d0ffb04763e9

**Comment by turansky** [comment]
Failed build expected

**Comment by Leonya** [comment]
Oops, fixed.

**Comment by turansky** [comment]
2 more `@Deprecated` annotations expected
For `render` and `hydrate` in `kotlin-react-dom-legacy`

**Pull request closed without merging**